pkg/cosign/cosign.go

```diff
@@ -50,37 +51,39 @@ func Initialize(client kubernetes.Interface, namespace, serviceAccount string, i
 }
 
 // VerifySignature verifies that the image has the expected key
-func VerifySignature(imageRef string, key []byte, repository string, log logr.Logger) (digest string, err error) {
-	pubKey, err := decodePEM(key)
+func VerifySignature(imageRef string, key string, repository string, log logr.Logger) (digest string, err error) {
+	ctx := context.Background()
+
+	var pubKey signature.Verifier
+
+	if strings.HasPrefix(key, "-----BEGIN PUBLIC KEY-----") {
+		pubKey, err = decodePEM([]byte(key))
+	} else {
+		pubKey, err = sigs.PublicKeyFromKeyRef(ctx, key)
+		if err != nil {
+			return "", errors.Wrap(err, "loading public key")
+		}
+	}
+
+	os.Setenv("COSIGN_REPOSITORY", repository)
```

Comment:
this line will fix the repository issue IMHO. PTAL @JimBugwadia @Dentrax @erkanzileli

Reply:
@JimBugwadia asked the following:
> The repository parameter allows using as separate Cosign repository. See: #2294. How will this be handled?

Kindly ping @dlorenc, would you mind sharing your thoughts? Will it fix this? thank you in advance.

Reply:
Not exactly. See: https://github.com/kyverno/kyverno/pull/2607#discussion_r740090961